android/coretest/src/androidTest/java/com/mongodb/stitch/android/core/StitchAppClientIntTests.kt

```diff
@@ -304,4 +308,39 @@ class StitchAppClientIntTests : BaseStitchAndroidIntTest() {
             )
         }
     }
+
+    @Test
+    fun testCustomFunctionLogin() {
+        val app = createApp()
+
+        val function = app.second.functions.create(FunctionCreator(
+            name = "funkyAuth",
+            source = """
+            exports = function(payload) {
+                return "foo";
+            };
+            """.trimIndent(),
+            canEvaluate = null,
+            private = false
+        ))
+
+        addProvider(app.second, ProviderConfigs.Function(function.id, function.name))
+
+        val client = getAppClient(app.first)
+
+        val user = Tasks.await(
+            client.auth.loginWithCredential(
+                FunctionCredential(Document("id", "123abc"))))
```

Comment:
[nit] do this in a `beforeEach` do each assertion below in its own it block with good description for each

Reply:
Not sure why this would be in a beforeEach block, we're only doing it in one integration test.